doc/feature/synapses.md

```diff
@@ -0,0 +1,215 @@
+    const uint32_t* postSegmentID() const;
+    const float* postDistance() const;
+    const float* postSurfacePositionX() const;
+    const float* postSurfacePositionY() const;
+    const float* postSurfacePositionZ() const;
+    const float* postCenterPositionX() const;
+    const float* postCenterPositionY() const;
+    const float* postCenterPositionZ() const;
+
+    const size_t* indices() const;
+    const float* conductance() const;
+    const float* utilization() const;
+    const float* depression() const;
+    const float* facilitation() const;
+    const float* decay() const;
+    const int* efficacy() const;
+};
+
+// sequentially & forwarding iteration through synapses for any number of GIDs
+class SynapsesStream
+{
+    bool eos() const;
+
+    // - async load of synapses for count cells
+    // - not order guarantee
+    // - result Synapses can be empty
+    // - throws if current + count leads to eos
+    std::future< Synapses > read( size_t count = 1 ) const;
+};
+
+```
+
+## File format
+
+Uses existing nrn* files
+
+## Examples
+
+### Stream synapse positions for pathway density voxelization
+
+```cpp
+const brain::Circuit circuit;
+const brain::SynapsesStream& stream =
+   circuit.getProjectedSynapses( circuit.getGIDs( "mtypetarget_1" ),
+                                 circuit.getGIDs( "mtypetarget_2" ));
+std::future< brain::Synapses > future = stream.read();
+while( !stream.eos( ))
+{
+    const brain::Synapses synapses = future.get();
+    if( !stream.eos( ))
+        future = stream.read(); // fetch next
+
+    const float* __restrict__ posx = synapses.preSurfacePositionX();
+    const float* __restrict__ posy = synapses.preSurfacePositionY();
+    const float* __restrict__ posz = synapses.preSurfacePositionZ();
+    for( size_t i = 0; i < synapses.size(); ++i )
+        doVoxelization( posx[i], posy[i], posz[i] );
+}
+```
+
+### Access all afferent synapses of a GIDSet
+
+```cpp
+const brain::Circuit circuit;
+const brain::Synapses& synapses =
+    circuit.getAfferentSynapses( circuit.getGIDs( "Layer1" ));
+for( const auto& synapse : synapses )
+    std::cout << synapse.postGID() << " " << synapse.decay() << std::endl;
+```
+
+### Retrograde projection
+
+```cpp
+const brain::Circuit circuit;
+const brion::GIDSet& preNeurons = circuit.getGIDs( "Layer1" );
+const brion::GIDSet postNeuron = { 1 };
+const brain::Synapses& synapses =
+    circuit.getProjectedSynapses( preNeurons, postNeuron );
+BOOST_CHECK( !synapses.empty( ));
+BOOST_CHECK_EQUAL( synapses.size(), 5 );
+```
+
+## Implementation
+
+* Data owned by container in the form of the arrays (transformed from
+  SynapseMatrix), loaded upon construction
+* Synapse object stores index to get return data from (parenting) container
+* SynapsesStream::read() uses async read of all synapse data of the given
+  fraction of GIDs
+* OPT: getProjectedSynapses() loads the synapses from the smaller GIDSet and
+  uses the correct side (afferent/efferent) accordingly
```

Comment:
Add dots to finish the sentences. I thought the first four lines where a paragraph in the beginning.